File: Rhino.Templates/content/VBCommand/MyCommand.1.vb
Imports System
Imports Rhino
Imports Rhino.Commands

Public Class MyCommand__1
    Inherits Command

    Shared _instance As MyCommand__1

    Public Sub New()
        ' Rhino only creates one instance of each command class defined in a
        ' plug-in, so it is safe to store a refence in a static field.
        _instance = Me
    End Sub

    '''<summary>The only instance of this command.</summary>
    Public Shared ReadOnly Property Instance() As MyCommand__1
        Get
            Return _instance
        End Get
    End Property

    '''<returns>The command name as it appears on the Rhino command line.</returns>
    Public Overrides ReadOnly Property EnglishName() As String
        Get
            Return "MyCommand.1"
        End Get
    End Property

    Protected Overrides Function RunCommand(ByVal doc As RhinoDoc, ByVal mode As RunMode) As Result

        ' TODO: complete command.
        Return Result.Success

    End Function
End Class

File: Rhino.Templates/content/CSCommand/MyCommand.1.cs
using System;
using Rhino;
using Rhino.Commands;

namespace MyNamespace
{
    public class MyCommand__1 : Command
    {
        public MyCommand__1()
        {
            Instance = this;
        }

        ///<summary>The only instance of the MyCommand command.</summary>
        public static MyCommand__1 Instance { get; private set; }

        public override string EnglishName => "MyCommand.1";

        protected override Result RunCommand(RhinoDoc doc, RunMode mode)
        {
            // TODO: complete command.
            return Result.Success;
        }
    }
}